### PR TITLE
removed logits casting in CrossEntropyLoss

### PR DESCRIPTION
**Pull request by shreydesai**
Summary:
Removes logits casting (to dtype32) in CrossEntropyLoss. This seems to be causing problems with the loss calculation for language modeling tasks. In particular, the losses are consistently higher as compared to the same model being trained with vanilla PyTorch.

Also, the open source implementation of CrossEntropyLoss does not perform casting by default: https://pytorch.org/docs/stable/_modules/torch/nn/functional.html.

Differential Revision: D15949540

### CODE REVIEW

**Comment by Titousensei** [comment]
Abandoned
